index.js

```diff
@@ -1 +1,100 @@
 // implement your API here
+const express = require('express');
+const db = require('./data/db')
+const server = express();
+const cors = require('cors')
+
+server.use(express.json())
+server.use(cors())
+
+server.get('/api/users', (req, res) => {
+    
+    db.find()
+        .then(allusers => {
+            res.json(allusers)
+        })
+        .catch(err => {
+            res.status(500).json({ err: "The users information could not be retrieved" })
+        });
+});
+
+server.get('/api/users/:id', (req, res) => {
+    
+    db.findById(req.params.id)
+        .then(user => {
+            if (user === undefined) {
+                res.status(404).json({ error: "The user with specified id does not exist" })
+            }
+            res.status(200).json(user)
+        })
+        .catch(err => {
+            res.status(500).json( err, "The users information could not be retrieved" )
+        });
+});
+
+server.post('/api/users', (req, res) => {
+    const { name, bio } = req.body
+    if (!name || !bio) {
+        res.status(400).json({ errorMessage: "Please provide name and bio for user" })
+    }
+    db.insert({
+        name,
+        bio,
+    })
+        .then(db.find()
```

Comment:
Make sure you pass an anonymous callback to your `.then` for it to work correctly. 👍
```js
.then(() => {
  db.find() // etc.
})
```